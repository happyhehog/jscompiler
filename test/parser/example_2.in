function sum(a, b, c) {
    return sum(a, b, 100) * c;
}

for (i = 1; i < 10; i++) {
    if (i === 5) break;
    console.log(i);
}