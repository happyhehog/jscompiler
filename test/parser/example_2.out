(program (sourceElements (sourceElement (functionDeclaration function sum ( (formalParameterList a , b , c) ) { (functionBody (sourceElements (sourceElement (statement (returnStatement return (expressionSequence (singleExpression (singleExpression (singleExpression sum) (arguments ( (argumentList (singleExpression a) , (singleExpression b) , (singleExpression (literal (numericLiteral 100)))) ))) * (singleExpression c))) (eos ;)))))) })) (sourceElement (statement (iterationStatement for ( (expressionSequence (singleExpression (singleExpression i) = (expressionSequence (singleExpression (literal (numericLiteral 1)))))) ; (expressionSequence (singleExpression (singleExpression i) < (singleExpression (literal (numericLiteral 10))))) ; (expressionSequence (singleExpression (singleExpression i) ++)) ) (statement (block { (statementList (statement (ifStatement if ( (expressionSequence (singleExpression (singleExpression i) === (singleExpression (literal (numericLiteral 5))))) ) (statement (breakStatement break (eos ;))))) (statement (expressionStatement (expressionSequence (singleExpression (singleExpression (singleExpression console) . (identifierName log)) (arguments ( (argumentList (singleExpression i)) )))))) (statement (emptyStatement ;))) })))))) <EOF>)